File: apps/frontend/src/lib/types/invoices.ts
export const ksefStatusSeverity = {
  not_sent: "warn",
  sent: "info",
  rejected: "danger",
  accepted: "success",
};

export const ksefStatusIcon = {
  not_sent: "pi pi-hourglass",
  sent: "pi pi-send",
  rejected: "pi pi-times",
  accepted: "pi pi-check-circle",
};

export const ksefStatusHint = {
  not_sent: "Nie wysłano do KSeF",
  sent: "W trakcie wysyłania do KSeF",
  rejected: "Faktura odrzucona przez KSeF",
  accepted: "Faktura obecna w systemie KSeF",
};

export const paymentType = {
  1: "Gotówka",
  2: "Karta",
  3: "Bon",
  4: "Czek",
  5: "Kredyt/Pożyczka",
  6: "Przelew",
  7: "Płatność mobilna",
};

export const paymentStatus = {
  paid: "Opłacono",
  partial: "Częściowo opłacono",
  unpaid: "Nie opłacono",
};

export const paymentStatusSeverity = {
  paid: "success",
  partial: "warn",
  unpaid: "danger",
};

export const paymentOptions = [
  { label: "Nie opłacono", value: "unpaid" },
  { label: "Opłacono", value: "paid" },
  { label: "Częściowo opłacono", value: "partial" },
];

export type invoiceTableType = "sales" | "purchases";

export interface InvoiceEntry {
  row_number: number;
  delivery_date: string | null;
  name: string;
  amount: number;
  unit: string;
  net_price: number;
  gross_price: number | null;
  net_total: number;
  tax_rate: number;
  tax_total: number | null;
  gross_total: number;
}

export interface InvoiceCompanyData {
  name: string;
  nip: string;
  address_l1: string;
  address_l2: string;
  email: string | null;
  phone: string | null;
}

export interface InvoicePayment {
  payment_status: string;
  payment_type: string;
  payment_date: string | null;
  payment_due_date: string | null;
  partial_payments: string | null;
  seller_bank_account_number: string | null;
  seller_bank_name: string | null;
}

export interface InvoiceResponse {
  id: string;
  invoice_number: string;
  invoice_type: string;
  ksef_number: string | null;
  invoicing_date: string | null;
  acquisition_date: string | null;
  permanent_storage_date: string | null;
  ksef_status: string;
  is_new: boolean;
  issue_date: string;
  issue_place: string;
  seller_info: InvoiceCompanyData;
  buyer_info: InvoiceCompanyData;
  entries: InvoiceEntry[];
  currency: string;
  net_total: number;
  tax_total: number;
  gross_total: number;
  payment: InvoicePayment;
  annotations: string | null;
  additional_info: string | null;
  footer_info: string | null;
  footer_registers: string | null;
}

File: apps/backend/backend/schemas/invoices.py
from pydantic import BaseModel
from datetime import datetime

from backend.db.models.invoices_payment import PaymentStatus, PaymentType
from backend.db.models.invoices_brief import KsefStatus
from backend.schemas.common import PageInfo


class InvoicesListResponse(BaseModel):
    """Response model for invoice list with pagination info."""

    invoices: list[dict]
    page_info: PageInfo


class InvoiceCompanyData(BaseModel):
    """Response model for invoice buyer and seller detals."""

    name: str
    nip: str
    address_l1: str | None
    address_l2: str | None
    email: str | None
    phone: str | None


class InvoiceEntry(BaseModel):
    """Response model for invoice list entry."""

    row_number: int
    delivery_date: str | datetime | None
    name: str
    amount: float
    unit: str
    net_price: float | None
    gross_price: float | None
    net_total: float | None
    tax_rate: int
    tax_total: float | None
    gross_total: float | None


class InvoicePayment(BaseModel):
    """Response model for invoice payment."""

    payment_status: PaymentStatus
    payment_type: PaymentType
    payment_date: str | datetime | None
    payment_due_date: str | datetime | None
    partial_payments: str | None
    seller_bank_account_number: str | None
    seller_bank_name: str | None


class InvoiceResponse(BaseModel):
    """Response model for full invoice with details."""

    id: str
    invoice_number: str
    invoice_type: str
    ksef_number: str | None
    invoicing_date: datetime | None
    acquisition_date: datetime | None
    permanent_storage_date: datetime | None
    ksef_status: KsefStatus
    is_new: bool
    ###
    issue_date: datetime
    issue_place: str | None
    ###
    seller_info: InvoiceCompanyData
    buyer_info: InvoiceCompanyData
    ###
    entries: list[InvoiceEntry]
    ###
    currency: str
    net_total: float
    tax_total: float
    gross_total: float
    ###
    payment: InvoicePayment | None
    ###
    annotations: str | None
    additional_info: str | None
    footer_info: str | None
    footer_registers: str | None


# temp models later merg with the upper ones 

class InvoiceEntryRequest(BaseModel):
    """Request model for invoice entry."""
    row_number: int
    delivery_date: str | None
    name: str
    amount: float
    unit: str
    net_price: float
    gross_price: float | None
    net_total: float
    tax_rate: str
    tax_total: float | None
    gross_total: float

class InvoiceCompanyDataRequest(BaseModel):
    """Request model for invoice buyer and seller detals."""
    name: str
    nip: str
    krs: str | None
    regon: str | None
    country_code: str
    address_l1: str | None
    address_l2: str | None
    address_correspondance_l1: str | None
    address_correspondance_l2: str | None
    email: str | None
    phone_number: str | None


class InvoicePaymentRequest(BaseModel):
    """Request model for invoice payment."""
    payment_status: str
    payment_type: int
    payment_date: str | None
    payment_due_date: str | None
    partial_payments: str | None
    seller_bank_account_number: str | None
    seller_bank_name: str | None


class InvoiceObjectRequest(BaseModel):
    """Request model for full invoice with details.+"""
    id: str | None
    invoice_number: str
    invoice_type: str
    ksef_number: str | None
    invoicing_date: str | None
    acquisition_date: str | None
    permanent_storage_date: str | None
    ksef_status: str
    is_new: bool
    issue_date: str
    issue_place: str
    seller_info: InvoiceCompanyDataRequest
    buyer_info: InvoiceCompanyDataRequest
    entries: list[InvoiceEntryRequest]
    currency: str
    net_total: float
    tax_total: float
    gross_total: float
    payment: InvoicePaymentRequest
    annotations: str | None
    additional_info: str | None
    footer_info: str | None
    footer_registers: str | None
